plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/DatasetTree/FileTree.tsx

```diff
@@ -23,13 +24,49 @@ const FileTree: React.FC<Props> = ({
   onDatasetAdd,
   onOpenDetails,
 }) => {
-  const [selectedID, select] = useState<string>();
+  const [selectedItem, selectItem] = useState<DataCatalogItem>();
   const [expandedFolders, setExpandedFolders] = useState<{ id?: string; name?: string }[]>([]);
 
-  const handleSelect = useCallback((dataID?: string) => {
-    select(dataID);
+  const handleSelect = useCallback((item?: DataCatalogItem) => {
+    selectItem(item);
   }, []);
 
+  useEffect(() => {
+    postMsg({ action: "catalogModalOpen" }); // Needed to trigger sending client storage data from Sidebar
+  }, []);
+
+  useEffect(() => {
+    const eventListenerCallback = (e: MessageEvent<any>) => {
+      if (e.source !== parent) return;
+      if (e.data.action === "catalogModalOpen") {
+        if (e.data.payload.expandedFolders) setExpandedFolders(e.data.payload.expandedFolders);
+        if (e.data.payload.dataset) {
+          const item = e.data.payload.dataset;
+          onOpenDetails?.(item);
+          handleSelect(item);
+          if (item.path) {
+            const expandedFolders = item.path.map((item: string) => ({ name: item }));
+            const index = expandedFolders.findIndex((folder: { id?: string; name?: string }) =>
+              folder.id ? folder.id === item.id : folder.name === item.name,
+            );
+            if (index >= 0) expandedFolders.splice(index, 1);
+            setExpandedFolders(expandedFolders);
+          }
+          setTimeout(() => {
+            postMsg({
+              action: "saveDataset",
+              payload: { dataset: undefined },
+            });
+          }, 500);
```

Comment:
I'm not sure we should remove the selected dataset with a timeout. Perhaps on modal close (in the sidebar.ts file) is better

Reply:
on modal close is not triggered when you click outside the modal, it is only triggered when you click on the close button.
However I removed the setTimeout and just left the postMsg method.
What do you think?

Reply:
I think our api has onClose where you can pass a callback